### PR TITLE
Initial Support for Mac Catalyst Platform

### LINKED ISSUE #31: enable Platforms 'macCatalyst' and 'driverKit' for Swift >= 5.5, Xcode >= 13
https://developer.apple.com/documentation/swift_packages/supportedplatform

### PR DESCRIPTION
**Pull request by mxcl**

### CODE REVIEW

**Comment by mxcl** [comment]
Basically I cannot verify this works without access to the macOS-11 image.

**Comment by chris-araman** [comment]
I can try it out on macos-11 and Xcode 13 beta. Out of town currently, so it might be a few days.

File: lib.ts [hunk 1]
```diff
@@ -218,6 +218,10 @@ export async function getDestination(platform: string) {
       return ['-destination', `id=${id}`]
     case 'macOS':
       return ['-destination', 'platform=macOS']
+    case 'mac-catalyst':
+      return ['-destination', "platform=macOS,variant='Mac Catalyst'"]
+    case 'driver-kit':
+      return ['-destination', "platform=macOS,variant='DriverKit'"]
```
Comment (chris-araman) [inline lib.ts:224]: I tried building CombineCloudKit for `mac-catalyst` with `swift: ~5.5` and `runs-on: macos-11`:
https://github.com/chris-araman/CombineCloudKit/pull/4/checks?check_run_id=2866696038

Looks like we need to use quote the arguments differently:
```
'platform=macOS,variant=Mac Catalyst'
'platform=macOS,variant=DriverKit'
```

I verified the quote changes (before and after) on my Mac, though not through GHA.

Comment (chris-araman) [inline lib.ts:224]: I tried building CombineCloudKit for `mac-catalyst` with `swift: ~5.5` and `runs-on: macos-11`:
https://github.com/chris-araman/CombineCloudKit/pull/4/checks?check_run_id=2866696038

Looks like we need to use quote the arguments differently:
```
'platform=macOS,variant=Mac Catalyst'
'platform=macOS,variant=DriverKit'
```

I verified the quote changes (before and after) on my Mac, though not through GHA.

**Review by chris-araman — CHANGES_REQUESTED** [review]

File: lib.ts (hunk 1 above)
Comment (mxcl) [reply lib.ts]: I adjusted the quoting

Comment (chris-araman) [reply lib.ts]: Thanks, @mxcl. I'd do it myself, but I don't have push access to this repo.

I must be overlooking something. 🤔 I'm still seeing [this](https://github.com/chris-araman/CombineCloudKit/runs/2870232843?check_suite_focus=true):
```
/Applications/Xcode_13.0.app/Contents/Developer/usr/bin/xcodebuild -destination "platform=macOS,variant='Mac Catalyst'" -scheme CombineCloudKit-Package OTHER_SWIFT_FLAGS=-warnings-as-errors build
```

...even when explicitly using your commit:
```
Download action repository 'mxcl/xcodebuild@0fdf692470bb67ae91616eaf09f787cf0fbca3a9'
```

I don't understand where those quotes are coming from, but they're causing `xcodebuild` to fail. Any ideas?

Comment (mxcl) [reply lib.ts]: Can't see why. I looked into child_process.spawn which is used here and it doesn’t auto quote. So… weird. If this command line instantiation works without quotes then I'm flummoxed.

However perhaps we need to be more specific? Using the `id` perhaps like we do for iOS etc.

Comment (chris-araman) [reply lib.ts]: Oh, I think I see the problem. 😉 

I'll see if I can figure out how to generate `index.js{.map}` on the fly.

**Review by chris-araman — APPROVED** [review]

**Comment by chris-araman** [comment]
Alright. I've [verified](https://github.com/chris-araman/CombineCloudKit/pull/4/checks?check_run_id=2871185799) `mac-catalyst` on `macos-11` with `swift: ~5.5` (Xcode 13 beta).

I've expanded the CI matrix here a bit, with appropriate (but unfortunate) `TODO`s and `continue-on-error` directives. I'm expecting GitHub to expand the audience of the `macos-11` runners, making them broadly available for some time before eventually updating the `macos-latest` tag. Maybe in time for macOS 12 to be released later this year? 😊

Another caveat: It seems that `mac-catalyst` requires code signing for the `test` action. I've changed the CI validation to use the `build` action. Perhaps it would be better to modify the action and emit a warning as we do for watchOS with Xcode < 12.5?

File: lib.ts (hunk 1 above)
Comment (chris-araman) [reply lib.ts]: > However perhaps we need to be more specific? Using the id perhaps like we do for iOS etc.

I don't think this is necessary, as the only differences seem to be in `arch`. I don't think there's a way to build for a destination with a different Mac Catalyst runtime (v13/v14) than what is included in the host macOS. Maybe there will be in the future.

**Comment by mxcl** [comment]
We have macOS-11 here now. I applied.

**Comment by mxcl** [comment]
> Perhaps it would be better to modify the action and emit a warning as we do for watchOS with Xcode < 12.5?

Agreed.

**Comment by mxcl** [comment]
https://github.com/mxcl/xcodebuild/pull/32/checks?check_run_id=2897121380

> Another caveat: It seems that mac-catalyst requires code signing for the test action.

`CODE_SIGNING=-` might work? or whatever it is you set for that.

**Comment by chris-araman** [comment]
> `CODE_SIGNING=-` might work? or whatever it is you set for that.

Good thinking. `CODE_SIGN_IDENTITY=-` gets the test target building and running.
https://github.com/mxcl/xcodebuild/pull/32/checks?check_run_id=2900523588

I've pulled the DriverKit stuff out of this PR, as I have been unsuccessful in getting it to build with Xcode 13 beta (1). Maybe we can try again in a later release.

I think this is ready to squash and merge, if you agree.

**Comment by mxcl** [comment]
Did a merge commit as history was neat

**Pull request merged**